traits/has_traits.py

```diff
@@ -2188,6 +2188,20 @@ def class_editable_traits ( cls ):
         return names
 
     class_editable_traits = classmethod( class_editable_traits )
+    
+    def visible_traits ( self ):
+        """Returns an alphabetically sorted list of the names of non-event
+        trait attributes associated with the current object, that should be GUI visible
+        """
+        return self.trait_names( type = not_event, editable = not_false, visible = not_false )
```

Comment:
"editable = not_false" is already the existing behavior of editable_traits, the only addition in the new visible_traits func was "visible = not_false"